src/base/http/connection.cpp

```diff
@@ -90,6 +90,18 @@ void Connection::read()
             }
             return;
 
+        case RequestParser::ParseStatus::NotImplemented: {
+                Logger::instance()->addMessage(tr("Unsupported Http request, closing socket. IP: %1")
+                    .arg(m_socket->peerAddress().toString()), Log::WARNING);
+
+                Response resp(501, "Not Implemented");
+                resp.headers[HEADER_CONNECTION] = "close";
+
+                sendResponse(resp);
+                m_socket->close();
```

Comment:
Maybe it should not close the socket yet, follow up requests might still be valid.

Reply:
That's fair, does the response need any headers in that case?

Reply:
I guess you still need `resp.headers[HEADER_CONNECTION] = "keep-alive";`